docusaurus/docs/create-a-plugin/extend-a-plugin/add-logs-metrics-traces-for-backend-plugins.md

```diff
@@ -0,0 +1,421 @@
+- Use a contextual logger whenever you have access to a `context.Context`.
+- Do not log sensitive information, such as data source credentials or IP addresses, or other personally identifiable information.
+
+#### Validate and sanitize input coming from user input
+
+If log messages or key-value pairs originate from user input they should be validated and sanitized. Be careful to not expose any sensitive information in log messages (secrets, credentials, and so on). It's especially easy to do by mistake when including a Go struct as a value.
+
+#### When to use which log level?
+
+- **Debug:** Informational messages of high frequency and/or less-important messages during normal operations.
+- **Info:** Informational messages of low frequency and/or important messages.
+- **Warning:** Should in normal cases not be needed. If used, it should be actionable.
+- **Error:** Error messages indicating some operation failed (with an error) and the program didn't have a way to handle the error.
+
+:::note
+
+Incoming requests of high frequency are normally more common for the `QueryData` endpoint, since--for example--the nature of a dashboard generates a request per panel or query.
+
+:::
+
+### Inspect logs locally
+
+Logs from a backend plugin are consumed by the connected Grafana instance and included in the Grafana server log.
+
+Each log message for a backend plugin will include a name, `logger=plugin.<plugin id>`. Example:
+
+```shell
+DEBUG[11-14|15:26:26] Debug msg     logger=plugin.grafana-helloworld-app someID=1
+INFO [11-14|15:26:26] Info msg      logger=plugin.grafana-helloworld-app queryType=default
+WARN [11-14|15:26:26] Warning msg   logger=plugin.grafana-helloworld-app key=value
+ERROR[11-14|15:26:26] Error msg     logger=plugin.grafana-helloworld-app error=An error occurred
+```
+
+You can enable [debug logging in your Grafana instance](https://grafana.com/docs/grafana/latest/troubleshooting/#troubleshoot-with-logs) and that will normally output a huge amount of information and make it hard to find the logs related to a certain plugin. However, using a named logger makes it convenient to enable debug logging only for a certain named logger and/or plugin:
+
+```
+[log]
+filters = plugin.<plugin id>:debug
+```
+
+Please refer to [Configure Grafana](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#log) for more details about setting up logging.
+
+Further, see [How to collect and visualize logs, metrics and traces](#collect-and-visualize-logs-metrics-and-traces).
+
+## Metrics
+
+Metrics are quantifiable measurements that reflect the health and performance of applications or infrastructure.
+
+Consider using metrics to provide real-time insight into the state of resources. If you want to know how responsive your plugin is or identify anomalies that could be early signs of a performance issue, metrics are a key source of visibility.
+
+### Metric types
```

Comment:
WDYT about giving a (brief) summary of the referenced material and then linked out for further reading. 

Even just expanding the acronym i.e. 
Rate (the number of requests per second)
Errors (the number of those requests that are failing)
Duration (the amount of time those requests take)

Reply:
I extended it a bit and decided to remove the RED reference altogether.